simplekv/__init__.py

```diff
@@ -181,6 +182,8 @@ def _check_valid_key(self, key):
 
         :param key: The key to be checked
         """
+        if not isinstance(key, text_type) and key is not None:
```

Comment:
Not here, I would say. There is a use-case with key-generating stores, but they overwrite `_put()` anyway.